gdeep/trainer/trainer.py

```diff
@@ -304,7 +320,23 @@ def closure() -> Tensor:
             pred, X, y = self._send_to_device(X, y)
             batch_metric = self.training_metric(pred, y)
             metric_list.append(batch_metric)
-            loss = self.loss_fn(pred, y)
+            if self.regularize == True:
+                loss = self.loss_fn(pred, y)
+                self.regularizer.regularizer_arguments["model"] = copy.deepcopy(
```

Comment:
Why is the deepcopy necessary here?

Reply:
This is purely esthetic but don't use `==True`. `self.regularize` is already a boolean

Reply:
right, it's not